sdk/src/client/stronghold/migration.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    num::NonZeroU32,
+    ops::Deref,
+    path::{Path, PathBuf},
+};
+
+use crypto::ciphers::chacha::{self};
+use iota_stronghold::{Client, SnapshotPath, Stronghold};
+use zeroize::Zeroize;
+
+use super::{common::PRIVATE_DATA_CLIENT_PATH, Error, StrongholdAdapter};
+use crate::client::stronghold::{check_or_create_snapshot, storage::insert as v3_insert};
+
+impl StrongholdAdapter {
+    /// Migrates a snapshot from version 2 to version 3.
+    pub fn migrate_snapshot_v2_to_v3<P: AsRef<Path>>(
```

Comment:
Can we instead add this as part of our migration logic?

Reply:
Not sure we can, we need bindings to this specific function